### PR TITLE
Resize mode based on bounding boxes

### PR DESCRIPTION
**Pull request by vanga**

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 1.36.0
```
Comment (rom1504) [inline HISTORY.md:1]: This will be added in another commit, no need to put it in this PR

File: img2dataset/resizer.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ class ResizeMode(Enum):
     keep_ratio = 1  # pylint: disable=invalid-name
     center_crop = 2  # pylint: disable=invalid-name
     border = 3  # pylint: disable=invalid-name
+    bounding_box = 4 # pylint: disable=invalid-name
```
[the remark this answers is not in the document]
Comment (vanga) [reply img2dataset/resizer.py:27]: thanks, let me think.

[the remark this answers is not in the document]
Comment (vanga) [reply img2dataset/resizer.py:27]: @rom1504 If we have to support both, we will need to introduce a separate flag I suppose instead of providing this new feature via resize_mode.
But, we may need to provide an option to specify the order of both of these operations then. 
I can imagine below scenario:
* someone downloads the initial dataset with one of the existing resize modes
* they do bounding box computations on top of resized image and prepares the metadata files for distributing

[the remark this answers is not in the document]
Comment (rom1504) [reply img2dataset/resizer.py:27]: I think it's clear the resizing should happen after the cutting in boxes

**Comment by rom1504** [comment]
you may be interested by https://github.com/rom1504/img2dataset/pull/241

**Comment by vanga** [comment]
@rom1504 w.r.t your below comment

> I think it's clear the resizing should happen after the cutting in boxes

The way one can prepare bbox metadata is to have existing data, that is possibly downloaded using this library. In which case images are already resized using existing modes and bounding boxes are computed based on the resized images.

So, when distributing the new dataset with bounding boxes, it is expected that the bounding box is done after the resizing.
It appears that laion400m, laion5b had some resize mode used.

Existing blur implementation is doing bounding box before resizing..

**Comment by rom1504** [comment]
I advise you preprocess the dataset you have to make the boxes fit the
original images
